lib/manager/pub/readme.md

```diff
@@ -0,0 +1,277 @@
+
+No
+
+---
+
+#### What format/syntax is the package file in? e.g. JSON, TOML, custom?
+
+YAML
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+Use YAML parser like [JS-YAML](https://github.com/nodeca/js-yaml)
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+Yes, `dependencies` and `dev-dependencies`.
+
+More details in: <https://www.dartlang.org/tools/pub/dependencies>
+
+Example from <https://github.com/brianegan/flutter_architecture_samples/blob/master/example/firestore_redux/pubspec.yaml>:
+
+```yaml
+environment:
+  sdk: '>=2.0.0 <3.0.0'
+
+dependencies:
+  meta: '>=1.1.0 <2.0.0'
+  redux: ^3.0.0
+  flutter_redux: ^0.5.0
+  flutter:
+    sdk: flutter
+  flutter_architecture_samples:
+    path: ../../
+  firebase_flutter_repository:
+    path: ../firebase_flutter_repository
+
+dev_dependencies:
+  test: ^1.3.0
+  mockito: ^3.0.0
+  flutter_driver:
+    sdk: flutter
+  flutter_test:
+    sdk: flutter
+  integration_tests:
+    path: ../integration_tests
+  todos_repository_flutter:
+    path: ../todos_repository_flutter
+```
+
+---
+
+#### List all the sources/syntaxes of dependencies that can be extracted:
+
+- [**SDK**](https://www.dartlang.org/tools/pub/dependencies#sdk)
+  ```yaml
+  dependencies:
+    flutter_driver:
+      sdk: flutter
+      version: ^0.0.1
+  ```
+
+- [**Version constraints**](https://www.dartlang.org/tools/pub/dependencies#version-constraints)
+  ```yaml
+  dependencies:
+    meta: '>=1.1.0 <2.0.0'
+    flutter_redux: 0.5.0
+  ```
+
+- [**Caret syntax**](https://www.dartlang.org/tools/pub/dependencies#caret-syntax)
+  ```yaml
+  dependencies:
+    redux: ^3.0.0
+  ```
+
+- [**Git packages**](https://www.dartlang.org/tools/pub/dependencies#git-packages)
+  ```yaml
+  dependencies:
+    kittens:
+      git: git://github.com/munificent/kittens.git
+  ```
+  ```yaml
+  dependencies:
+    kittens:
+      git: git@github.com:munificent/kittens.git
+  ```
+  ```yaml
+  dependencies:
+    kittens:
+      git:
+        url: git://github.com/munificent/kittens.git
+        ref: some-branch
+  ```
+  ```yaml
+  dependencies:
+    kittens:
+      git:
+        url: git://github.com/munificent/cats.git
+        path: path/to/kittens
+  ```
+
+- [**Path packages**](https://www.dartlang.org/tools/pub/dependencies#path-packages)
+  ```yaml
+  dependencies:
+    transmogrify:
+      path: /Users/me/transmogrify
+  ```
+
+- [**Hosted packages**](https://www.dartlang.org/tools/pub/dependencies#hosted-packages)
+  ```yaml
+  dependencies:
+    transmogrify:
+      hosted:
+        name: transmogrify
+        url: http://your-package-server.com
+      version: ^1.4.0
+  ```
+
+---
+
+#### Describe which types of dependencies above are supported and which will be implemented in future:
+
+All
```

Comment:
Potentially we implement using basic semver first, then custom hosts after that, etc.